esphome/components/display/display_buffer.cpp

```diff
@@ -773,12 +773,29 @@ Color Animation::get_grayscale_pixel(int x, int y) const {
   return Color(gray, gray, gray, alpha);
 }
 Animation::Animation(const uint8_t *data_start, int width, int height, uint32_t animation_frame_count, ImageType type)
-    : Image(data_start, width, height, type), current_frame_(0), animation_frame_count_(animation_frame_count) {}
-int Animation::get_animation_frame_count() const { return this->animation_frame_count_; }
+    : Image(data_start, width, height, type),
+      current_frame_(0),
+      animation_frame_count_(animation_frame_count),
+      loop_start_frame_(0),
+      loop_end_frame_(animation_frame_count_),
+      loop_count_(0),
+      loop_current_iteration_(1) {}
+uint32_t Animation::get_animation_frame_count() const { return this->animation_frame_count_; }
 int Animation::get_current_frame() const { return this->current_frame_; }
+void Animation::set_loop(uint32_t start_frame, uint32_t end_frame, int count) {
+  loop_start_frame_ = std::min(start_frame, animation_frame_count_);
+  loop_end_frame_ = std::min(end_frame, animation_frame_count_);
+  loop_count_ = count;
```

Comment:
I just noticed that you should probably reset the current loop iteration when updating the parameters, for example via a lambda.
```suggestion
  if (loop_count_ != count) {
    this->loop_current_iteration_ = 1;
  }
  loop_count_ = count;
```

Reply:
The lambda thought is a valid one; I hadn't thought about that, I considered this setter method as "part of the constructor".

Nevertheless I would do that without the if; when the parameters are reset in a lambda, the iteration count should also restart.

Also, weren't you the one (rightly) warning me about useless loops? In your suggestion there is a useless assignment: if the ˋifˋ is not entered, there is no point in setting the ˋloop_countˋ to ˋcountˋ; they are already equal ;-)